src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -167,6 +177,79 @@ private void addDependsOnIfCatalogObjectModelExistOnTaskVariable(
+
+        if (task.getPreScript() != null && task.getPreScript().getScriptUrl() != null &&
+            !task.getPreScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, task.getPreScript().getScriptUrl().toString());
+        }
+
+        if (task.getPostScript() != null && task.getPostScript().getScriptUrl() != null &&
+            !task.getPostScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, task.getPostScript().getScriptUrl().toString());
+        }
+
+        if (task.getFlowScript() != null && task.getFlowScript().getScriptUrl() != null &&
+            !task.getFlowScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, task.getFlowScript().getScriptUrl().toString());
+        }
+
+        if (task.getCleaningScript() != null && task.getPreScript().getScriptUrl() != null &&
+            !task.getCleaningScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, task.getCleaningScript().getScriptUrl().toString());
+        }
+
+        if (task.getForkEnvironment() != null && task.getForkEnvironment().getEnvScript() != null &&
+            task.getForkEnvironment().getEnvScript().getScriptUrl() != null &&
+            !task.getForkEnvironment().getEnvScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, task.getForkEnvironment().getEnvScript().getScriptUrl().toString());
+        }
+
+        if (task instanceof ScriptTask && ((ScriptTask) task).getScript() != null &&
+            ((ScriptTask) task).getScript().getScriptUrl() != null &&
+            !((ScriptTask) task).getScript().getScriptUrl().toString().isEmpty()) {
+            addDependsOn(keyValueMapBuilder, ((ScriptTask) task).getScript().getScriptUrl().toString());
+        }
+
+        if (task.getSelectionScripts() != null) {
+            task.getSelectionScripts().forEach(selectionScript -> {
+                if (selectionScript.getScriptUrl() != null && !selectionScript.getScriptUrl().toString().isEmpty()) {
+                    addDependsOn(keyValueMapBuilder, selectionScript.getScriptUrl().toString());
+                }
+            });
+        }
+
+    }
+
+    private void addDependsOn(ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String scriptUrl) {
+
+        keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(getNameAndBucketFromUrl(scriptUrl),
+                                                               getRevisionFromUrl(scriptUrl).orElse(LATEST_VERSION),
+                                                               ATTRIBUTE_DEPENDS_ON_LABEL));
+
+    }
+
+    private String getNameAndBucketFromUrl(String scriptUrl) {
+        Matcher matcher = URL_PATTERN.matcher(scriptUrl);
+        if (!(scriptUrl.matches(SCRIPT_URL_REGEX) && matcher.find())) {
```

Comment:
I don't think you should throw an exception here, but rather, only add depends on when the pattern matches

Reply:
Ok